### DISCUSSION TITLE
Find nested model objects by _id (works with name, but not the id)

### DISCUSSION THREAD

**Discussion by diegocgaona** [BeanieODM/beanie#914] [Question] [answered]
I have 2 models:

```python
class Drinks(Document):
    id: PydanticObjectId = Field(default_factory=PydanticObjectId, alias="_id")
    name: Indexed(str, unique=True)
    drinks_tags: list[DrinksTags] | None = None

class DrinksTags(Document):
    id: PydanticObjectId = Field(default_factory=PydanticObjectId, alias="_id")
    name: Indexed(str, unique=True)
    color: str | None = None
```

I want to find all `Drinks  `which have the `DrinkTags `with an specific ID. I tried this:

```python
all_drinks = await Drinks.find({"drinks_tags._id": item_id}).to_list()
```

But it doesn't work.

It works if I try for example:
```python
all_drinks = await Drinks.find({"drinks_tags.name": item_name}).to_list()

````
My `item_id `is an `PydanticObjectId `, but it can't find any item.

Why it works with the `item_name`, but not the `_id`?

What I want when I update a tag, to get all the drinks which have the specific tag, and update then too, but I can't find the drinks with then... I know it could be possible if I get a list with all the drinks and do some for loops to find and update then, but I think it would be more efficient and natural to do it with mongodb operations.

Thanks, in advance.

**Comment by hgalytoby** [comment, marked as answer]
I've tested this query with no problems.

```py
print(await Drinks.find({"drinks_tags._id": PydanticObjectId('661df66e53b0ca2825058801')}).to_list()) print(await Drinks.find({"drinks_tags._id": (await Drinks.find_one()).drinks_tags[0].id}).to_list())
```

### output
```
[
Drinks(
  id=ObjectId('661df66e53b0ca2825058802'), 
  revision_id=None, 
  name='123',
  drinks_tags=[
    DrinksTags(
      id=ObjectId('661df66e53b0ca2825058801'), 
      revision_id=None, 
      name='123',
      color=None
    )
  ]
)
]

[
Drinks(
  id=ObjectId('661df66e53b0ca2825058802'), 
  revision_id=None, 
  name='123',
  drinks_tags=[
    DrinksTags(
      id=ObjectId('661df66e53b0ca2825058801'), 
      revision_id=None, 
      name='123',
      color=None
    )
  ]
)
]
```

Suggestion, I see you use Document for both models, I think you should use Link instead.

https://beanie-odm.dev/tutorial/relations/

**Comment by diegocgaona** [comment]
Hi @hgalytoby 

I found the problem... my data was fault. I don't know why (appears to be ok) but there was something wrong with the id.

I'm not using link because of some things I read on the MongoDB documentation.
I'm using embedded documents because in this case they are small and when I read the Drink, the tags data will always be show together, and the update of the tags would be rare (then I usually, will no need to perform this action of find and update). 
They say it will be better for performance (one read operation).

In another case, with a related collection more complex and which can be update more frequently, I will use Link and BackLinks.

Thanks!!
